src/libexpr/eval.cc

```diff
@@ -1370,7 +1365,31 @@ void EvalState::autoCallFunction(Bindings & args, Value & fun, Value & res)
             if (j != args.end()) {
                 actualArgs->attrs->push_back(*j);
             } else if (!i.def) {
-                throwTypeError("cannot auto-call a function that has an argument without a default value ('%1%')", i.name);
+                throwUndefinedVarError(i.pos, R"(cannot auto-call a function that has an argument without a default value ('%1%')
+
+An 'auto-call' is when a nix expression is evaluated without any external arguments.
```

Comment:
This is not correct. An auto-call is when a top-level function is called *with* external arguments *or* with default values if no matching argument is provided.

It's probably better to say something like `you need to pass "--arg %1% ...'  or '--argstr %1% ...'".

Reply:
`UndefinedVarError` is probably misleading. There is no undefined variable, just a missing argument.